File: 프로그래머스/~2021/위장.js
function solution(clothes) {
    var map = {};
    for (var i = 0; i < clothes.length; i++) {
        var k = clothes[i][1];
        map[k] = (map[k] || 1) + 1;
    }
    console.log(map)
    var re = 1;
    for(var key in map){
        re *= map[key];
    }

    return re - 1;
}

File: 프로그래머스/~2021/위장.java
import java.util.*;
class Solution {
    HashMap<String, Integer> map = new HashMap<>();
    public int solution(String[][] clothes) {
        for(String[] cl : clothes){
            String k = cl[1];
            String v = cl[0];
            
            map.put(k, map.getOrDefault(k, 0) + 1);
        }
        
        int mul = 1;
        for(String k : map.keySet()){
            int v = map.get(k);
            mul *= (v + 1);
        }
        
        return mul - 1;
    }
}